nomos-services/storage/src/api/chain/requests.rs

```diff
@@ -0,0 +1,99 @@
+use nomos_core::header::HeaderId;
+use tokio::sync::oneshot::Sender;
+use tracing::error;
+
+use crate::{
+    api::{chain::StorageChainApi, StorageApiRequest, StorageBackendApi, StorageOperation},
+    backends::StorageBackend,
+    StorageMsg, StorageServiceError,
+};
+
+pub enum ChainApiRequest<B: StorageBackend> {
+    GetBlock {
+        header_id: HeaderId,
+        response_tx: Sender<Option<<B as StorageChainApi>::Block>>,
+    },
+    StoreBlock {
+        header_id: HeaderId,
+        block: <B as StorageChainApi>::Block,
+    },
+}
+
+impl<B> StorageOperation<B> for ChainApiRequest<B>
+where
+    B: StorageBackend + StorageBackendApi,
+{
+    async fn execute(self, backend: &mut B) -> Result<(), StorageServiceError<B>> {
+        match self {
+            Self::GetBlock {
+                header_id,
+                response_tx,
+            } => handle_get_block(backend, header_id, response_tx).await,
+            Self::StoreBlock { header_id, block } => {
+                handle_store_block(backend, header_id, block).await
+            }
+        }
+    }
+}
+
+async fn handle_get_block<B: StorageBackend>(
+    backend: &mut B,
+    header_id: HeaderId,
+    response_tx: Sender<Option<B::Block>>,
+) -> Result<(), StorageServiceError<B>> {
+    let result = backend.get_block(header_id).await.map_err(|e| {
+        error!("Failed to get block: {:?}", e);
+
+        let key = "header_id".to_owned().into();
```

Comment:
Same point as for the DA operations about unifying the definitions and using enums where it would make more sense.